### DISCUSSION TITLE
How to debug compents while running tests in VS code?

### DISCUSSION THREAD

**Discussion by mberrg** [vitest-dev/vitest#2874] [Q&A]
I'm having problem debugging components with Vitest, I'm not sure if this is bug or bad configuration. 

For this demo I'm using the Hello.vue from [vitest/examples](https://github.com/vitest-dev/vitest/tree/main/examples/vue) and test basic.test.ts.

In vs-code I've added this launch setting: 
``` JSON
{
        "type": "node",
        "request": "launch",
        "trace": true,
        "name": "Debug Current Test File",
        "autoAttachChildProcesses": true,
        "disableOptimisticBPs": "true",
        "skipFiles": ["<node_internals>/**", "**/node_modules/**"],
        "program": "${workspaceRoot}/node_modules/vitest/vitest.mjs",
        "args": ["--inspect-brk", "run", "${relativeFile}"],
        "smartStep": true,
        "console": "integratedTerminal"
}

```
None of configs or files from the Vitest example is modified. 

In Hello.vue I add a breakpoint at line 6 (the ref). Then I launch the debugger and all is good, and it stops at correct line :-)

However Hello.vue is simple, and as soon as I add a e.g. a style to the component, the breakpoint stops at incorrect line number ( unbounded ). See below for the modified component, the breakpoint is still set to line 6, but it stops at line 12.

``` VUE
<script setup lang="ts">
import { computed, ref } from 'vue'

const props = defineProps<{ count: number }>()

const times = ref(2)
const result = computed(() => props.count * times.value)

defineExpose(props)
</script>

<template>
  <div class="fixed-width">{{ count }} x {{ times }} = {{ result }}</div>
  <button @click="times += 1">
    x1
  </button>
</template>
<style scoped lang="css">
.fixed-width {
  width: 540px;
}
</style>
```

It seems that there is something odd with the source map?

I've tried to add `sourcemap: 'inline'` to vitest.config.ts, however it does not change behaviour. I tested The VS code debugger with  Vite and Chrome at this hits correct line number. This issue is only happening during Vitest

Does any one have a valid setup/configuration to debug components during vitest?

**Comment by AriPerkkio** [comment]
Vue's source maps are incorrect. This is also causing coverage issues for `.vue` files.

There are some similar reports on https://github.com/vuejs/vue-loader/issues/1778 but I was unable to get this working using the work-around mentioned on that issue. If you find out the root cause or solution for this please let me know. 

<img height="320px" src="https://user-images.githubusercontent.com/14806298/219109774-d8ab9919-9b82-49ad-96a9-e59ea26ca075.png" />

<img height="320px" src="https://user-images.githubusercontent.com/14806298/219109920-b511bb72-1b8f-4744-9348-7ffec57bf451.png" />

<img height="320px" src="https://user-images.githubusercontent.com/14806298/219110061-d9ae482a-366e-4eea-89e6-8a5cd01fb6f9.png" />

**Comment by bonham** [comment]
I have exactly the same problem: https://github.com/vitest-dev/vitest/issues/5380
